pliers/extractors/base.py

```diff
@@ -132,98 +170,103 @@ def history(self):
+            'filename', 'history', and 'source_file'. Note that these values
+            are often long strings, so the returned DF will be considerably
+            larger.
+        extractor_names (str, bool): How to handle extractor names when
+            returning results. The specific behavior depends on whether format
+            is 'long' or 'wide'. Valid values include:
+
+                - 'prepend': In both 'long' and 'wide' formats, feature names
+                  will be prepended with the Extractor name (e.g.,
+                  "FaceExtractor#face_likelihood").
+                - 'drop' or False: In both 'long' and 'wide' formats, extractor
+                  names will be omitted entirely from the result. Note that
+                  this can create feature name conflicts when merging results
+                  from multiple Extractors, so is generally discouraged.
+                - 'column': In 'long' format, extractor name will be included
+                  as a separate column. Not valid for 'wide' format (and will
+                  raise an error).
+                - 'multi': In 'wide' format, a MultiIndex will be used for the
+                  columns, with the first level of the index containing the
+                  Extractor name and the second level containing the feature
+                  name. This value is invalid if format='long' (and will raise
+                  and error).
+                - True: When format='long', behaves like 'column'. When
+                  format='wide', behaves like 'prepend'.
+
+        aggfunc (str, Callable): If format='wide' and extractor_names='drop',
+            it's possible for name clashes between features to occur. In such
+            cases, the aggfunc argument is passed onto pandas' pivot_table
+            function, and specifies how to aggregate multiple values for the
+            same index. Can be a callable or any string value recognized by
+            pandas. By default (None), 'mean' will be used for numeric columns
+            and 'first' will be used for object/categorical columns.
+
+    Returns: a pandas DataFrame. For format details, see 'format' argument.
     '''
 
-    # Flatten list recursively
-    results = flatten(results)
-
-    stims = defaultdict(list)
-
-    for r in results:
-        stims[hash(r.stim)].append(r)
-
-    # First concatenate all features separately for each Stim
-    for k, v in stims.items():
-        stims[k] = ExtractorResult.merge_features(v, **merge_feature_args)
-
-    # Now concatenate all Stims
-    stims = list(stims.values())
-    return stims[0] if len(stims) == 1 else \
-        ExtractorResult.merge_stims(stims)
+    _timing = True if timing == 'auto' else timing
+
+    if extractor_names is True:
+        extractor_names = 'prepend' if format == 'wide' else 'column'
+
+    dfs = [r.to_df(timing=_timing, metadata=metadata, format='long',
+                   extractor_name=True, add_object_id=True)
```

Comment:
Shouldn't we make `add_object_id` a method argument and just pass it through here? On a related note, worth considering whether to do the same "auto" business here when all `object_id`s are 0.

Reply:
Good idea. I'll add an argument for that, and make the default 'auto'.